modules/core/src/main/java/org/apache/ignite/internal/client/GridClientClusterState.java

```diff
@@ -17,21 +17,40 @@
 package org.apache.ignite.internal.client;
 
 import java.util.UUID;
+import org.apache.ignite.cluster.ClusterState;
 
 /**
  *  Interface for manage state of grid cluster and obtain information about it: ID and tag.
  */
 public interface GridClientClusterState {
     /**
      * @param active {@code True} activate, {@code False} deactivate.
+     * @deprecated Use {@link #state()} instead.
      */
+    @Deprecated
     public void active(boolean active) throws GridClientException;
 
     /**
      * @return {@code Boolean} - Current cluster state. {@code True} active, {@code False} inactive.
+     * @deprecated Use {@link #state(ClusterState)} instead.
      */
+    @Deprecated
     public boolean active() throws GridClientException;
 
+    /**
+     * @return Current cluster state.
+     * @throws GridClientException If the request to get the cluster state failed.
+     */
+    public ClusterState state() throws GridClientException;
+
+    /**
+     * Changes cluster state to {@code newState}.
+     *
+     * @param newState New cluster state.
+     * @throws GridClientException If the request to change the cluster state failed.
+     */
+    public void state(ClusterState newState) throws GridClientException;
```

Comment:
may be something like : **changeState** would be informative here?

Reply:
Any objections to the current naming?

Reply:
yes

Reply:
Let's keep this conversation for the core reviewer.